keel-schema-generator/src/test/kotlin/com/netflix/spinnaker/keel/schema/GeneratorTests.kt

```diff
@@ -688,14 +687,46 @@ internal class GeneratorTests {
         )
     }
 
-    @Test
-    fun `generic sub-types contain all of the base type properties and their own`() {
-      expectThat(schema.`$defs`["${Bar::class.java.simpleName}${Foo::class.java.simpleName}"])
-        .isA<AllOf>()
-        .get { allOf }
-        .one { isA<Reference>().get { `$ref` } isEqualTo "#/\$defs/${Foo::class.java.simpleName}" }
-        .one { isA<ObjectSchema>().get { properties.keys }.containsExactly(Foo<*>::genericProperty.name) }
-    }
+    @TestFactory
+    fun genericSubTypes() =
+      listOf(Bar::class.java, Baz::class.java)
+        .flatMap { subType ->
+          val subTypeName = "${subType.simpleName}${Foo::class.java.simpleName}"
+
+          listOf(
+            dynamicTest("generic sub-type $subTypeName contains all of the base type properties") {
+              expectThat(schema.`$defs`[subTypeName])
+                .isA<AllOf>()
+                .get { allOf }
+                .one { isA<Reference>().get { `$ref` } isEqualTo "#/\$defs/${Foo::class.java.simpleName}" }
+            },
+
+            dynamicTest("generic sub-type $subTypeName contains the generic properties and discriminator") {
+              expectThat(schema.`$defs`[subTypeName])
+                .isA<AllOf>()
+                .get { allOf }
+                .one {
+                  isA<ObjectSchema>().get { properties.keys }
+                    .containsExactlyInAnyOrder(
+                      Foo<*>::type.name,
+                      Foo<*>::genericProperty.name,
+                    )
+                }
+            },
+
+            dynamicTest("generic sub-type discriminator in $subTypeName is a single value enums") {
```

Comment:
If it's a single value, why does it need to be an `enum`?

Reply:
Because that's how JSON schema represents that. You can't specify a string literal like you can in Cue.

Reply:
Huh, actually draft 6 introduced a `const` keyword for this. I'm making some changes to how this works anyway so I'll work that in.